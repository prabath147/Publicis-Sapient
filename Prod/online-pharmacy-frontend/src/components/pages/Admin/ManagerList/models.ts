import { PageableResponse } from "../storeList/models";

export interface Manager 
{
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: Date;
  approvalStatus: "APPROVED" | "PENDING" | "REJECTED";
}


export interface ManagerListResponse extends PageableResponse{
  data:Manager[]
}