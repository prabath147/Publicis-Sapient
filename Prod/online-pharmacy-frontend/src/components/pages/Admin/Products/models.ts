
export interface CategorySet {
  categoryId: number;
  categoryName: string;
}
export interface Product {
  categorySet: CategorySet[];
  description: string;
  dosageForm: string;
  imageUrl: string;
  productId: number;
  productName: string;
  productType: boolean;
  proprietaryName: string;
  quantity: number;
}

