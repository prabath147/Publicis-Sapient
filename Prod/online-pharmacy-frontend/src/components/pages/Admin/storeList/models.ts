import { Item } from "../../Manager/models";

export interface Test {
  data: Store[];
  pageNumber: number;
  pageSize: number;
  totalRecords: number;
  totalPages: number;
  isLastPage: boolean;
}


export interface Store {
  storeId: number;
  storeName: string;
  address: Address;
  createdDate: string;
  manager: Manager;
  revenue: number;
}

export interface Address {
  addressId: number;
  street: string;
  city: string;
  state: string;
  pinCode: number;
  country: string;
}

export interface Manager {
  managerId: number;
  name: string;
  phoneNo: string;
  licenseNo: string;
  registrationDate: string;
  approvalStatus: string;
}

export interface PageableResponse{
  "pageNumber":number,
  "pageSize":number,
  "totalRecords":number,
  "totalPages":number,
  "isLastPage":boolean
}

export interface getStoreInventoryResponse extends PageableResponse{
  data : Item[]
}


