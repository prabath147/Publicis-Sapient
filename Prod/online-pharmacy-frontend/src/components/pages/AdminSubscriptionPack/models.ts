export interface Subscriptionv{
    subscriptionId:number,
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit
    subscriptionCost:number,
    period:number,
    status:string
}

export interface Benefit{
    delivery_discount:number,
    one_day_delivery:boolean
}
export interface SubDetails{
    subscriptionName:string,
    description:string,
    startDate:Date,
    endDate:Date,
    benefits:Benefit,
    subscriptionCost:number,
    period:number
}

export interface SubRequest{
    name:string,
    description:string,
    stDate:Date,
    edDate:Date,
    deliveryDiscount:number,
    oneDayDelivery:string,
    cost:number,
    period:number
}