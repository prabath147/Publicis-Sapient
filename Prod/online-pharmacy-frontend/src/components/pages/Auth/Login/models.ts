export interface User {
  id: number;
  username: string;
  email: string;
  role: string;
}

export interface LoginRequest {
  username: string;
  password: string;
}

export interface LoginResponse {
  id: number;
  username: string;
  email: string;

  token: string;
  refreshToken: string;
}

export interface tokenStructure {
  sub: string;
  role: string;
  iat: number;
  exp: number;
  id:number;
}
