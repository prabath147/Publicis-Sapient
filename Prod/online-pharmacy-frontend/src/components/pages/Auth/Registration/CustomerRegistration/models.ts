import { AddressInterface } from './../../../../ui/forms/AddressForm';

export interface CustomerRegistrationRequest {
  username: string;
  email: string;
  password: string;
  role: ["customer"];

  detailObject: {
    fullName: string;
    mobileNumber: number;
    address: AddressInterface
  };
}

export interface CustomerRegistrationForm {
  fullName: string;
  username: string;
  mobileNumber: number;
  email: string;
  password: string;
  confirmPassword: string;
  address: AddressInterface
}
