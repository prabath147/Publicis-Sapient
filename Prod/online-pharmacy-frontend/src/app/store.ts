import { Action, configureStore, ThunkAction } from "@reduxjs/toolkit";
import StoreReducer from "../components/pages/Admin/storeList/StoreSlice";
import SubscriptionReducer from "../components/pages/AdminSubscriptionPack/SubscriptionvSlice";
import UserReducer from "../components/pages/Auth/Login/UserSlice";
import CartReducer from "../components/pages/Cart/CartSlice";
import ManagerReducer from "../components/pages/Manager/ManagerSlice";
import NotificationReducer from "../components/pages/Open/notificationList/NotificationSlice";
// import OrderSlice from "../components/pages/orderDetail/OrderSlice";
// import OrderListSlice from "../components/pages/orderList/OrderListSlice";

export const store = configureStore({
  reducer: {
    storeL: StoreReducer,
    user: UserReducer,
    notification: NotificationReducer,
    viewsubscription: SubscriptionReducer,
    // addsubscription: AddSubcriptionRe,
    cart: CartReducer,
    // order: OrderSlice,
    // orderL: OrderListSlice,
    manager: ManagerReducer,
  },
  middleware: (getDefaultMiddleware) =>
    getDefaultMiddleware({
      serializableCheck: false,
    }),
});

export type AppDispatch = typeof store.dispatch;
export type RootState = ReturnType<typeof store.getState>;
export type AppThunk<ReturnType = void> = ThunkAction<
  ReturnType,
  RootState,
  unknown,
  Action<string>
>;
