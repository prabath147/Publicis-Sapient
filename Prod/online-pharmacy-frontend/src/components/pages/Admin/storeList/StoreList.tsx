import {
  ActionIcon, Card, Container, Divider,
  Group, Pagination,
  Table
} from "@mantine/core";
import { showNotification } from "@mantine/notifications";
import { Dispatch } from "@reduxjs/toolkit";
import { IconChevronRight } from "@tabler/icons";
import { useEffect, useState } from "react";
import { Link } from "react-router-dom";
import { useAppDispatch, useAppSelector } from "../../../../app/hooks";
import { Store } from "./models";
import { getStoreList } from "./StoreAPI";
import { loadStoreData, selectStores } from "./StoreSlice";

export default function StoreList() {
  const sList = useAppSelector(selectStores);
  const dispatch = useAppDispatch();
  const [activePage, setPage] = useState(1);
  const [pageCount, setPageCount] = useState(1);

  const limit = 10;
  useEffect(() => {
    dispatch(loadDataAction());
    // eslint-disable-next-line react-hooks/exhaustive-deps
  }, [activePage]);

  const loadDataAction = () => async (dispatch: Dispatch) => {
    try {
      const payload = await getStoreList(activePage - 1, limit);
      const stores: Store[] = [];
      setPageCount(payload.data.totalPages);
      payload.data.data.forEach((element: Store) => {
        stores.push(element);
      });
      dispatch({
        type: loadStoreData,
        payload: stores,
      });
    } catch (error) {
      showNotification({ message: "Oops Something went Wrong", color: "red" })
      // alert(error);
    }
  };

  const rows = sList.map((row: Store) => (
    <tr>
      <td>{row.storeId}</td>
      <td>{row.manager.managerId}</td>
      <td>{row.storeName}</td>
      <td>{row.address.city}</td>
      <td>{row.createdDate.toString().split("T")[0]}</td>
      <td>{row.revenue}</td>
      <td>
        <Link to={"/store-details/" + row.storeId} style={{ textDecoration: 'none' }}>
          <ActionIcon variant='subtle' color={"blue"}>
            <IconChevronRight size={26} />
          </ActionIcon>
        </Link>
      </td>
    </tr>
  ));

  return (

    <div>
      <Card shadow="sm" p="lg" radius="md" withBorder>
        <Container>
          <h1>List of Stores</h1>
          <Table highlightOnHover>
            <thead>
              <tr>
                <th>Store Id</th>
                <th>Manager Id</th>
                <th>Store Name</th>
                <th>City</th>
                <th>Registration Date</th>
                <th>Revenue</th>
              </tr>
            </thead>
            <tbody>{rows}</tbody>
          </Table>
          <Divider my="sm" />

          <Group position="center">
            <Pagination
              position="center"
              page={activePage}
              onChange={setPage}
              total={pageCount}
              radius="lg"
            />

          </Group>
        </Container>

      </Card>
    </div>

  );
}