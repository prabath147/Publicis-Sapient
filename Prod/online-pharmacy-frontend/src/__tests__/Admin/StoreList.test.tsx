import { Provider } from 'react-redux';
import renderer from 'react-test-renderer';
import { store } from '../../app/store';
import { Store } from '../../components/pages/Admin/storeList/models';
import StoreList from '../../components/pages/Admin/storeList/StoreList';
import UserReducer, { loadStoreData, StoresState } from "../../components/pages/Admin/storeList/StoreSlice";

it('component created', () => {
    const component = renderer.create(
        <Provider store={store}>
            <StoreList />
        </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('component data loaded', () => {
    const component = renderer.create(
        <Provider store={store}>
            <StoreList />
        </Provider>
    )
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
})

it('test fetch', async () => {
    const data = {
        data: [
            {
                id: 1,
                store_manger_id: 101,
                store_name: "pharma",
                address: 'xyz..',
                reg_date: new Date("2022-08-16"),
                store_desc: "abc...",
            },
        ],
    };
});

const initialState: StoresState = {
    stores: [],
};

it("should handle initial state", () => {
    expect(UserReducer(undefined, { type: "unknown" })).toEqual({
        stores: [],
    });
});


it("should handle data load", () => {
    const data: Store[] = [
        {
            storeId: 1,
            managerId: 101,
            storeName: "pharma",
            address: "xyz..",
            createdDate: new Date("2022-08-16"),
            revenue: 2000
        },
    ];
    const actual = UserReducer(initialState, loadStoreData(data));
    expect(actual.stores).toEqual(data);
});
