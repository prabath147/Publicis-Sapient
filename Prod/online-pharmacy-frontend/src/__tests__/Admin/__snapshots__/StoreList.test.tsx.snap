// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`component created 1`] = `
<div>
  <div
    className="mantine-Paper-root mantine-Card-root mantine-ergsfy"
  >
    <div
      className="mantine-Container-root mantine-15pjuqq"
    >
      <h1>
        List of Stores
      </h1>
      <table
        className="mantine-Table-root mantine-b146xb"
        data-hover={true}
      >
        <thead>
          <tr>
            <th>
              Store Id
            </th>
            <th>
              Manager Id
            </th>
            <th>
              Store Name
            </th>
            <th>
              City
            </th>
            <th>
              Registration Date
            </th>
            <th>
              Revenue
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
      <div
        className="mantine-Divider-root mantine-Divider-horizontal mantine-polw48"
        role="separator"
      />
      <div
        className="mantine-Group-root mantine-1u7f61y"
      >
        <div
          className="mantine-Group-root mantine-Group-child mantine-hsvael"
          role="navigation"
        >
          <button
            aria-disabled={true}
            className="mantine-Group-child mantine-Pagination-item mantine-18brkxg"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height={16}
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.219 8l3.3 3.3-.943.943L5.333 8l4.243-4.243.943.943-3.3 3.3z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            aria-current="page"
            aria-label={null}
            className="mantine-Group-child mantine-Pagination-item mantine-18brkxg"
            data-active={true}
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            1
          </button>
          <button
            aria-disabled={true}
            className="mantine-Group-child mantine-Pagination-item mantine-18brkxg"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height={16}
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.781 8l-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943 3.3-3.3z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`component data loaded 1`] = `
<div>
  <div
    className="mantine-Paper-root mantine-Card-root mantine-ergsfy"
  >
    <div
      className="mantine-Container-root mantine-15pjuqq"
    >
      <h1>
        List of Stores
      </h1>
      <table
        className="mantine-Table-root mantine-b146xb"
        data-hover={true}
      >
        <thead>
          <tr>
            <th>
              Store Id
            </th>
            <th>
              Manager Id
            </th>
            <th>
              Store Name
            </th>
            <th>
              City
            </th>
            <th>
              Registration Date
            </th>
            <th>
              Revenue
            </th>
          </tr>
        </thead>
        <tbody />
      </table>
      <div
        className="mantine-Divider-root mantine-Divider-horizontal mantine-polw48"
        role="separator"
      />
      <div
        className="mantine-Group-root mantine-1u7f61y"
      >
        <div
          className="mantine-Group-root mantine-Group-child mantine-hsvael"
          role="navigation"
        >
          <button
            aria-disabled={true}
            className="mantine-Group-child mantine-Pagination-item mantine-18brkxg"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height={16}
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M7.219 8l3.3 3.3-.943.943L5.333 8l4.243-4.243.943.943-3.3 3.3z"
                fill="currentColor"
              />
            </svg>
          </button>
          <button
            aria-current="page"
            aria-label={null}
            className="mantine-Group-child mantine-Pagination-item mantine-18brkxg"
            data-active={true}
            onClick={[Function]}
            tabIndex={0}
            type="button"
          >
            1
          </button>
          <button
            aria-disabled={true}
            className="mantine-Group-child mantine-Pagination-item mantine-18brkxg"
            disabled={true}
            onClick={[Function]}
            type="button"
          >
            <svg
              fill="none"
              height={16}
              viewBox="0 0 16 16"
              width={16}
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8.781 8l-3.3-3.3.943-.943L10.667 8l-4.243 4.243-.943-.943 3.3-3.3z"
                fill="currentColor"
              />
            </svg>
          </button>
        </div>
      </div>
    </div>
  </div>
</div>
`;
